cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -25,6 +26,12 @@
 def remove_installed_state():
     remove_state('kubernetes-worker.components.installed')
 
+@hook('stop')
+def remove_this_node():
+    ''' Deletes this unit's node when the unit is destroyed. '''
+    kubectl = ['kubectl', '--kubeconfig=/srv/kubernetes/config']
+    command = kubectl + ['delete', 'node', gethostname()]
```

Comment:
We kind of need to use that though. There's no other way that I know of right now to extrapolate routable hostnames other than to use the hostname itself.

And this is with the assumption the cloud is setup in such a way that hostnames are resolvable (most major clouds are doing this).